integration/src/integration/java/pl/allegro/tech/hermes/integration/management/ListUnhealthySubscriptionsForOwnerBasedOnGraphiteTest.java

```diff
@@ -0,0 +1,387 @@
+        );
+
+        // then
+        assertThat(listUnhealthySubscriptionsDisrespectingSeverity("Team A")).contains(
+                new UnhealthySubscription("ownedSubscription1", topic.getQualifiedName(), Severity.CRITICAL,
+                        ImmutableSet.of(malfunctioning(11, subscription1.getQualifiedName().toString()))),
+                new UnhealthySubscription("ownedSubscription2", topic.getQualifiedName(), Severity.IMPORTANT,
+                        ImmutableSet.of(malfunctioning(11, subscription2.getQualifiedName().toString()))),
+                new UnhealthySubscription("ownedSubscription3", topic.getQualifiedName(), Severity.NON_IMPORTANT,
+                        ImmutableSet.of(malfunctioning(11, subscription3.getQualifiedName().toString())))
+        );
+        assertThat(listUnhealthySubscriptionsDisrespectingSeverityAsPlainText("Team A")).isEqualTo(
+                "ownedSubscription1 - Consuming service returns a lot of 5xx codes for subscription "
+                        + subscription1.getQualifiedName().toString() + ", currently 11 5xx/s\r\n"
+                        + "ownedSubscription2 - Consuming service returns a lot of 5xx codes for subscription "
+                        + subscription2.getQualifiedName().toString() + ", currently 11 5xx/s\r\n"
+                        + "ownedSubscription3 - Consuming service returns a lot of 5xx codes for subscription "
+                        + subscription3.getQualifiedName().toString() + ", currently 11 5xx/s"
+        );
+    }
+
+    @Test
+    public void shouldTimeoutUnhealthySubscriptionsRequest() {
+        // given
+        Topic topic = apiOperations.buildTopic(randomTopic("group", "topic").build());
+        createSubscriptionForOwner(topic, "ownedSubscription1", "Team A", Severity.CRITICAL);
+        int graphiteDelay = 1000;
+        graphiteEndpoint.returnMetricWithDelay(subscriptionMetricsStub(subName(topic, "ownedSubscription1"))
+                .withRate(50)
+                .withStatusRate(200, 11).build(), graphiteDelay);
+
+        // when
+        long start = System.currentTimeMillis();
+        Throwable thrown = catchThrowable(() -> listUnhealthyAsPlainText("Plaintext", "Team A", true, Lists.asList(), Lists.asList()));
+        long end = System.currentTimeMillis();
+
+        // then
+        assertThat(thrown).isInstanceOf(InternalServerErrorException.class);
+        assertThat(end - start < graphiteDelay);
+
+    }
+
+    @Test
+    public void shouldReportSuspendedSubscriptionAsHealthy() {
+        // given
+        Topic topic = apiOperations.buildTopic(randomTopic("group", "topic").build());
+        Subscription s = createSubscriptionForOwner(topic, "subscription1", "Team A");
+
+        // when
+        s.setState(Subscription.State.SUSPENDED);
+
+        // then
+        assertThat(listUnhealthySubscriptionsForOwner("Team A")).isEmpty();
```

Comment:
**question:** Can we verify that the suspended subscription is healthy (rather than verifying that there are no unhealthy ones)?

Reply:
This tests is just a copy paste from already [existing](https://github.com/allegro/hermes/blob/1c22e61ae5e2b3842f56cb6c8c50808b71a8e2b0/integration/src/integration/java/pl/allegro/tech/hermes/integration/management/ListUnhealthySubscriptionsForOwnerTest.java#L313) tests for `UnhealthyEndpoint` class. It was renamed to indicate we use Graphite underneath because by default we use Prometheus right now. When it comes to testing `UnhealthyEndpoint` I think it properly checks if there are no unhealthy subscriptions returned.

Reply:
right now we don't have any cache metrics. but we should probably add a few in the future.